influxdb/_dataframe_client.py

```diff
@@ -202,7 +202,8 @@ def _to_dataframe(self, rs, dropna=True):
             df = pd.DataFrame(data)
             df.time = pd.to_datetime(df.time)
             df.set_index('time', inplace=True)
-            df.index = df.index.tz_localize('UTC')
+            if df.index.tzinfo is None:
+                df.index = df.index.tz_localize('UTC')
```

Comment:
I am wondering what was the intent of the `tz_localize` in the original code:
- ensure there is a timezone defined
- ensure the timezone is UTC
If it is the latter we should probably add:
```suggestion
                df.index = df.index.tz_localize('UTC')
            else:
                df.index = df.index.tz_convert('UTC')
```

Can anyone check what was the behavior of `tz_localize` on an index with a timezone with an old version of panda? Does it convert to the requested timezone or do nothing?

Reply:
Older version of pandas (that work with influxdb-python) already had the same behavior of raising an error:

```python
>>> import pandas as pd
>>> pd.__version__
u'0.22.0'
>>> pd.DatetimeIndex(['2018-01-01T00:00+1'], tz='CET').tz_localize('UTC')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/olojkine/Developpement/ps-custom/bhc/python/vpp/commitments/venv/lib/python2.7/site-packages/pandas/util/_decorators.py", line 118, in wrapper
    return func(*args, **kwargs)
  File "/Users/olojkine/Developpement/ps-custom/bhc/python/vpp/commitments/venv/lib/python2.7/site-packages/pandas/core/indexes/datetimes.py", line 1851, in tz_localize
    raise TypeError("Already tz-aware, use tz_convert to convert.")
TypeError: Already tz-aware, use tz_convert to convert.
```

Reply:
The change happened in `to_datetime` (on line 203)

https://pandas.pydata.org/pandas-docs/stable/whatsnew/v0.24.0.html#parsing-datetime-strings-with-timezone-offsets

It used to generate timezone-naive Timestamps, and now it keeps the timezone from the original string.

Reply:
I suggest merging the PR as-is. It has the same behavior on old and new versions of pandas.

Reply:
Thanks!
Then I guess this code never encountered dates it had to convert and the suggested `else` branch is not necessary.
And the if branch will probably never be taken with 0.24.0, but should be kept for compatibility with older versions of pandas.